src/test/integration/IntegrationBase.t.sol

```diff
@@ -0,0 +1,367 @@
+        for (uint i = 0; i < strategies.length; i++) {
+            assertEq(prevShares[i] + addedShares[i], curShares[i], err);
+        }
+    }
+
+    /// @dev Check that the staker has `removedShares` prior shares for each
+    /// strategy since the last snapshot
+    function assert_Snap_RemovedStakerShares(
+        User staker, 
+        IStrategy[] memory strategies, 
+        uint[] memory removedShares,
+        string memory err
+    ) internal {
+        uint[] memory curShares = _getStakerShares(staker, strategies);
+        // Use timewarp to get previous staker shares
+        uint[] memory prevShares = _getPrevStakerShares(staker, strategies);
+
+        // For each strategy, check (prev - removed == cur)
+        for (uint i = 0; i < strategies.length; i++) {
+            assertEq(prevShares[i] - removedShares[i], curShares[i], err);
+        }
+    }
+
+    function assert_Snap_IncreasedQueuedWithdrawals(
+        User staker, 
+        IDelegationManager.Withdrawal[] memory withdrawals,
+        string memory err
+    ) internal {
+        uint curQueuedWithdrawals = _getCumulativeWithdrawals(staker);
+        // Use timewarp to get previous cumulative withdrawals
+        uint prevQueuedWithdrawals = _getPrevCumulativeWithdrawals(staker);
+
+        assertEq(prevQueuedWithdrawals + withdrawals.length, curQueuedWithdrawals, err);
+    }
+
+    function assert_Snap_IncreasedTokenBalances(
+        User staker,
+        IERC20[] memory tokens,
+        uint[] memory addedTokens,
+        string memory err
+    ) internal {
+        uint[] memory curTokenBalances = _getTokenBalances(staker, tokens);
+        // Use timewarp to get previous token balances
+        uint[] memory prevTokenBalances = _getPrevTokenBalances(staker, tokens);
+
+        for (uint i = 0; i < tokens.length; i++) {
+            uint prevBalance = prevTokenBalances[i];
+            uint curBalance = curTokenBalances[i];
+
+            assertEq(prevBalance + addedTokens[i], curBalance, err);
+        }
+    }
+
+    /**
+     * Helpful getters:
+     */
+
+    /// @dev For some strategies/underlying token balances, calculate the expected shares received
+    /// from depositing all tokens
+    function _calculateExpectedShares(IStrategy[] memory strategies, uint[] memory tokenBalances) internal returns (uint[] memory) {
+        uint[] memory expectedShares = new uint[](strategies.length);
+
+        for (uint i = 0; i < strategies.length; i++) {
+            IStrategy strat = strategies[i];
+
+            uint tokenBalance = tokenBalances[i];
+            if (strat == BEACONCHAIN_ETH_STRAT) {
+                // TODO - need to calculate this
+                // expectedShares[i] = eigenPodManager.underlyingToShares(tokenBalance);
+                revert("_calculateExpectedShares: unimplemented for native eth");
+            } else {
+                expectedShares[i] = strat.underlyingToShares(tokenBalance);
+            }
```

Comment:
pretty sure we just want to set `expectedShares[i] = tokenBalances[i]` in this case, but perhaps it depends a bit on the exact handling we use for 'beaconChainETH' elsewhere

Reply:
i think some of the spots where i revert with "unimplemented for native eth" type things have easy answers, but i didn't want to leave some functional paths in when most of it doesnt work